components/base/kinematicbase/kinematics.go

```diff
@@ -102,56 +107,106 @@ func (ddk *differentialDriveKinematics) GoToInputs(ctx context.Context, desired
 	if err != nil {
 		return err
 	}
-	validRegion, err := ddk.newValidRegionCapsule(current, desired)
+	distErr, headingErr, err := ddk.errorState(current, desired)
+	if err != nil {
+		return err
+	}
+	validRegion, err := ddk.newValidRegionCapsule(current, desired, distErr)
 	if err != nil {
 		return err
 	}
 
+	logger := golog.NewDebugLogger("kinematics")
+
+	t := time.NewTimer(timeout)
+
+	positionChange := make(chan bool)
+	movementErr := make(chan error)
+
 	// this loop polls the error state and issues a corresponding command to move the base to the objective
 	// when the base is within the positional threshold of the goal, exit the loop
-	for err = ctx.Err(); err == nil; err = ctx.Err() {
-		col, err := validRegion.CollidesWith(spatialmath.NewPoint(r3.Vector{X: current[0].Value, Y: current[1].Value}, ""))
-		if err != nil {
-			return err
-		}
-		if !col {
-			return errors.New("base has deviated too far from path")
-		}
+	go func() {
```

Comment:
This is a no-no in production code. If something in your goroutine panicked, it would crash all of RDK.

To solve this we have this function which you can pass this codeblock directly into: https://github.com/viamrobotics/goutils/blob/b45e9e59763295615c25debe7861f69b55d0f52a/runtime.go#L141